### PR TITLE
ref(ts) Convert ProjectBadge to Typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import styled from 'react-emotion';
 import overflowEllipsis from 'app/styles/overflowEllipsis';
 import space from 'app/styles/space';
 
-const BadgeDisplayName = styled('span')<{hideOverflow?: boolean}>`
+const BadgeDisplayName = styled('span')<{hideOverflow?: string | boolean}>`
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.tsx]: @markstory When would `hideOverflow` be a string? And what values would it be? I hope it's not like `'false'` which is truthy.

**Review by scefali — APPROVED** [review]

File: src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx [hunk 2]
```diff
@@ -3,15 +3,25 @@ import PropTypes from 'prop-types';
...
 
-export default class ProjectBadge extends React.PureComponent {
+type Props = {
+  project: Project;
+  avatarSize: number;
+  className?: string;
+  hideAvatar?: boolean;
+  hideName?: boolean;
+  description?: React.ReactNode;
+  avatarClassName?: string | React.CSSProperties;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx]: When would this be `CSSProperties`?

Comment (markstory) [reply src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx]: I thought was the right type to use for style attributes based on `BaseBadge` having a proptype of `string, object` and the resulting prop ending up in a `css` prop of `StyledAvatar`.

**Review by billyvg — APPROVED** [review]

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Comment by dashed** [comment]
@markstory What was the state of this?

**Comment by markstory** [comment]
There are a few build test errors with this and I haven't had time to circle back around to it.

**Comment by markstory** [comment]
Closing this. I'll make another pass at it soon.

**Pull request closed without merging**